File: 1-twoSum/twoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
    // use dictionary to reduce searching time.
        
        int[] rslt = new int[2];
        
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i< nums.length; i++) {
        if (map.containsKey(target - nums[i])) {
            rslt[0] = i;
            rslt[1] = map.get(target - nums[i]);
            
        } else {
            map.put(nums[i], i);
        }
        }
        
        return rslt;

    }
}

File: 1-twoSum/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int,int> mp;  //num->idx
        for(int i=0;i<nums.size();i++){
            if(mp.find(nums[i])!=mp.end()){
                res.push_back(mp[nums[i]]);
                res.push_back(i);
            }
            else{
                mp[target-nums[i]]=i;
            }
        }
        return res;
    }
};